### ISSUE TITLE
Archive instead of purge

### ISSUE THREAD

**Issue by marshall-mcmullen** [sphinx-contrib/confluencebuilder#697]
Greetings,

I really like the feature provided via `confluence_purge` and `confluence_purge_from_root` in order to ensure our published docs have the same structure as what we expect and delete any pages that shouldn't be there.

The problem is that at our company we don't have permission to delete pages. Instead we archive them. This way they can be viewed in the confluence archive and can be retrieved later, etc. But they effectively are no longer in the tree of documents. 

Is this something you could look into?

Thank you so much!

**Comment by jdknight** [comment]
Tried to implement such a capability in #699. In theory, users would flag for legacy pages to be archived instead of purged.

Would you mind testing it out and seeing if it works for in your environment. I have only done some limited testing at this time. The change set can be installed using the following command:

```
pip install git+https://github.com/sphinx-contrib/confluencebuilder.git@support-archiving-pages
```

And use the following configuration option over purging instead:

```
confluence_cleanup_archive = True
```

**Comment by jdknight** [comment]
Additional sanity checks have been performed to help verify the initial support for this capability. These changes were merged into the main branch and have been included in the recent v1.9 release (now available on PyPI). With this feature being available on PyPI, will preemptively close this issue. If any specific issues are observed with archiving, we can re-open this issue or new issues outlining specific issues with archiving can be made. A note that since Atlassian has this API marked as experimental, a users experience may change over time is the API evolves. This notice is added in the configuration notes in the documentation.

**Comment by marshall-mcmullen** [comment]
@jdknight wow, that was fast! I am so happy to see that you already had done some work on this in the past. Thank you very much for getting this merged into an actual release. I will update the version we use internally this week and let you know if there are any questions or problems. But on the surface it looks perfect. Thank you so much !!
